### DISCUSSION TITLE
Image/File are not deleted from the supabase storage bucket

### DISCUSSION THREAD

**Discussion by jigarc007** [supabase/supabase#30928] [Questions]
my bucket structure is like below
====================================
user-uploaded/{{userId}}/{{imageName}}

my API looks like this 
=====================================
  const { error, data } = await supabase.storage
      .from('user-uploaded')
      .remove([imagePath]);
      
I'm getting no error but the data returns an empty array []
The image path is also correct 

when I'm checking the getPuplicURL image URL is also receives 
 const { data: fileData, error: fileError } = await supabase.storage
  .from('user-uploaded')
  .getPublicUrl(imagePath);
  
then also Image are not deleted  from the supabae storage 
could you help me resolve the issue?

**Comment by GaryAustin1** [comment]
getPublicUrl does not actually go to the server it just does string concatenation.  It will work for any filename whether it exists or not.

Do you meet Select and Delete RLS for that bucket on storage.objects (see the Storage UI)?  Both are required to delete the file even for a public bucket.
